### PR TITLE
Fix issues with LVM mounting.

### PR DESCRIPTION
**Pull request by agibbons27**
Fixed issue with mounting LVMs, specifically for mounting LVMs using a network block device. Also added a regular expression to remove non-numeric characters when parsing volume size from `lvdisplay` command output.

### CODE REVIEW

File: imagemounter/filesystems.py [hunk 1]
```diff
@@ -636,14 +636,18 @@ def mount(self):
         os.environ['LVM_SUPPRESS_FD_WARNINGS'] = '1'
 
         # find free loopback device
-        self._find_loopback()
+        # No need to find loopback device if using nbd device
+        if "nbd" not in self.volume.get_raw_path():
+            self._find_loopback()
+        else:
+            self.loopback = ""
```
Comment (ralphje) [inline imagemounter/filesystems.py]: I'd prefer to set it to None, in other words, to just remove the else statement here.

File: imagemounter/filesystems.py [hunk 2]
```diff
@@ -636,14 +636,18 @@ def mount(self):
         os.environ['LVM_SUPPRESS_FD_WARNINGS'] = '1'
 
         # find free loopback device
-        self._find_loopback()
+        # No need to find loopback device if using nbd device
+        if "nbd" not in self.volume.get_raw_path():
```
Comment (ralphje) [inline imagemounter/filesystems.py]: Is this case also true for other block devices? If the LVM is already mounted on another loopdevice for instance.

File: imagemounter/volume.py [hunk 3]
```diff
@@ -95,6 +95,10 @@ def fstype(self):
     def mountpoint(self):
         return getattr(self.filesystem, 'mountpoint', None)
 
+    @mountpoint.setter
+    def mountpoint(self, new_mountpoint):
+        self.filesystem.mountpoint = new_mountpoint
+
```
Comment (ralphje) [inline imagemounter/volume.py]: Is this really necessary? Can you provide an example for this?

File: imagemounter/volume_system.py [hunk 4]
```diff
@@ -533,7 +533,7 @@ def detect(self, volume_system, vstype='detect'):
                 cur_v.info['label'] = line.replace("LV Name", "").strip()
             if "LV Size" in line:
                 size, unit = line.replace("LV Size", "").strip().split(" ", 1)
-                cur_v.size = int(float(size.replace(',', '.')) * {'KiB': 1024, 'MiB': 1024 ** 2,
+                cur_v.size = int(float(re.sub(r'[^0-9.]', "", size.replace(',', '.'))) * {'KiB': 1024, 'MiB': 1024 ** 2,
```
Comment (ralphje) [inline imagemounter/volume_system.py:536]: Do you perhaps have an example line where this fails? It's OK if you haven't but it would be helpful for some tests in the future.

**Review by ralphje — CHANGES_REQUESTED** [review]

**Comment by ralphje** [comment]
For LVM, maybe we should make sure that we also detect the line lvm may spit out that it is using another device for detecting the LVM.

File: imagemounter/volume.py (hunk 3 above)
Comment (agibbons27) [reply imagemounter/volume.py]: This was part of some testing I was doing but is no longer needed. Removed in most recent commit.

File: imagemounter/volume_system.py (hunk 4 above)
Comment (agibbons27) [reply imagemounter/volume_system.py:536]: In some cases, `lvm lvdisplay` may output other characters in the LV Size section.

Example:
`LV Size                <19.00 GiB`

In this case, `<` is included in the `size` variable and would throw an error when being converted to a float. The regular expression removes that character so no error occurs.

File: imagemounter/filesystems.py (hunk 1 above)
Comment (agibbons27) [reply imagemounter/filesystems.py]: Updated in most recent commit.

**Comment by ralphje** [comment]
This seems OK to me, but I have two questions:

* You removed the 0-offset check. I question whether this is correct in the case that two LVMs are present on the same disk, and one is located in offset 0.
* Is nbd the only device that has the characteristic to be detectable without loopback? Isn't  this the case for all images mounted as block devices in /dev/?

Perhaps the 0-check should be combined with the block device check in here.

**Comment by agibbons27** [comment]
I'm not sure if that applies to other types of devices. The specific image I'm using for testing is a .vmdk image that I mount with the qemu-nbd method. Once mounted, the volume group gets activated and the volumes can be mounted. If the 0-check for offset is included, the volume is not mounted.

I could just be misunderstanding this, but I'm not sure what the purpose of the 0-check is. Maybe this is just an edge case, but for my test image, the 0-check prevents mounting the LVM.
